File: Daily/1846.maximum-element-after-decreasing-and-rearranging.js
/*
 * @lc app=leetcode id=1846 lang=javascript
 *
 * [1846] Maximum Element After Decreasing and Rearranging
 */

// @lc code=start
/**
 * @param {number[]} arr
 * @return {number}
 */
var maximumElementAfterDecrementingAndRearranging = function (arr) {
    arr.sort((a, b) => a - b);

    let max = 0;

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] > max) {
            max++;
        }
    }

    return max;
};
// @lc code=end

File: src/2026/1846.maximum-element-after-decreasing-and-rearranging.cpp
/*
 * @lc app=leetcode id=1846 lang=cpp
 *
 * [1846] Maximum Element After Decreasing and Rearranging
 */

#include <algorithm>
#include <vector>

// @lc code=start
class Solution {
 public:
  int maximumElementAfterDecrementingAndRearranging(std::vector<int>& arr) {
    std::ranges::sort(arr);
    arr[0] = 1;
    for (int i = 1; i < arr.size(); i++) {
      arr[i] = std::min(arr[i - 1] + 1, arr[i]);
    }
    return arr.back();
  }
};
// @lc code=end
